wolfssl/ssl.h

```diff
@@ -30,6 +30,7 @@
 /* for users not using preprocessor flags*/
 #include <wolfssl/wolfcrypt/settings.h>
 #include <wolfssl/version.h>
+#include <wolfssl/wolfcrypt/types.h> /* for XFILE */
```

Comment:
A goal of our main public facing header is to keep this as opaque as possible.  That is, hide our different types / ifdefs / options as much as possible.  

For OpenSSL compatibility we already have a pattern to deal with FILE*, like this: 

#if !defined(NO_FILESYSTEM) && !defined(NO_STDIO_FILESYSTEM)
/* wolfSSL extension, provide last error from SSL_get_error
   since not using thread storage error queue */
#include <stdio.h>
WOLFSSL_API void  wolfSSL_ERR_print_errors_fp(FILE*, int err);

Can we use the same pattern here instead?  XFILE shouldn't be needed here I wouldn't think.